<launch>
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find exploration)/cfg/custom_rosconsole.config"/>

    <arg name="robot_name" default="robot1"/>
    <arg name="eta" value="1.0"/>
    <arg name="Geta" value="15.0"/>

    <arg name="map_topic" default="map_continuity"/>

    <group ns="$(arg robot_name)">
        <node pkg="rrt_exploration" type="global_rrt_detector" name="global_detector" output="screen">
            <param name="eta" value="$(arg Geta)"/>
            <param name="map_topic" value="$(arg map_topic)"/>
        </node>
    
        <node pkg="rrt_exploration" type="local_rrt_detector" name="local_detector" output="screen">
            <param name="eta" value="$(arg eta)"/>
            <param name="map_topic" value="$(arg map_topic)"/>
            <param name="robot_frame" value="$(arg robot_name)/base_link"/>
        </node>
        
        <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
            <remap from="move_base_node/global_costmap/costmap" to="move_base/global_costmap/costmap"/>
            <param name="map_topic" value="$(arg map_topic)"/>
            <param name="info_radius" value="1"/> 
            <param name="costmap_clearing_threshold" value="70"/> 
            <param name="goals_topic" value="/detected_points"/>
            <param name="namespace_init_count" value="1"/>
            <param name="namespace" value="/robot"/> 
            <param name="n_robots" value="1"/>
            <param name="rate" value="100"/>
        </node>
        
        <node pkg="rrt_exploration" type="assigner.py" name="assigner" output="screen">
            <remap from="move_base_node/NavfnROS/make_plan" to="move_base/NavfnROS/make_plan"/>
            <param name="map_topic" value="$(arg map_topic)"/>
            <param name="global_frame" value="$(arg robot_name)/map"/>
            <param name="info_radius" value="1"/> 
            <param name="info_multiplier" value="3.0"/> 
            <param name="hysteresis_radius" value="3.0"/> 
            <param name="hysteresis_gain" value="2.0"/> 
            <param name="frontiers_topic" value="/$(arg robot_name)/filtered_points"/> 
            <param name="n_robots" value="1"/>
            <param name="namespace_init_count" value="1"/>
            <param name="namespace" value="/robot"/>
            <param name="delay_after_assignement" value="10.0"/>
            <param name="rate" value="100"/>
        </node>

        <!-- exploration support -->
        <include file="$(find exploration)/launch/includes/exploration_support.launch.xml">
            <arg name="robot_name" value="$(arg robot_name)"/>
            <arg name="use_rosbag" value="false"/>
            <arg name="frontier_detection" value="false"/>
            <arg name="branch_detection" value="false"/>
            <arg name="map_fill" value="false"/>
            <arg name="orig_map" value="map_continuity"/>
            <arg name="fill_map" default="fill_map"/>
        </include>

        <!-- rosbag -->
        <include file="$(find exploration_support)/launch/rosbag_record.launch">
            <arg name="topics" value="/$(arg robot_name)/map_continuity 
                                        /$(arg robot_name)/pose
                                        /$(arg robot_name)/pose_log/$(arg robot_name)
                                        /$(arg robot_name)/scan
                                        /$(arg robot_name)/move_base/global_costmap/costmap
                                        /$(arg robot_name)/rtabmap/mapPath
                                        /tf
                                        /tf_static"/>
            <arg name="compress" value="true"/>
	<arg name="name" value="rosbag_rrt"/>
        </include>

        <!-- rosbag -->
        <include file="$(find exploration_support)/launch/rosbag_record.launch">
            <arg name="topics" value="/$(arg robot_name)/map_continuity"/>
            <arg name="compress" value="true"/>
            <arg name="name" value="rosbag_rrt_filtered"/>
        </include>
    </group>
  
</launch>
